### PR TITLE
[#1401] feat:(dashboard): Support deleting lost shuffle servers

### PR DESCRIPTION
**Pull request by yl09099**
### What changes were proposed in this pull request?

Support deleting lost shuffle servers:
If some shuffle servers are lost, they need to go offline and the Coordinator has been in the lost state. 
Add the delete function to delete these servers in the memory.

image

### Why are the changes needed?

Fix: #1401 

### Does this PR introduce _any_ user-facing change?

Yes.

### How was this patch tested?

Add UT.

### CODE REVIEW

**Comment by yl09099** [comment]
@xianjingfeng @rickyma @jerqi @maobaolong PATL

**Comment by rickyma** [comment]
ping @xianjingfeng ~

File: coordinator/src/main/java/org/apache/uniffle/coordinator/web/resource/ServerResource.java [hunk 1]
```diff
@@ -199,6 +200,16 @@ public Response<Map<String, Integer>> getNodeStatusTotal() {
...
   }
 
+  @DELETE
+  @Path("/deleteServer")
+  public Response<String> deleteLostedServer(@QueryParam("serverId") String serverId) {
+    ClusterManager clusterManager = getClusterManager();
+    if (clusterManager.deleteLostServerById(serverId)) {
+      return Response.success("success");
+    }
+    return Response.success("failed");
```
Comment (xianjingfeng) [inline coordinator/src/main/java/org/apache/uniffle/coordinator/web/resource/ServerResource.java:210]: This is a bit weird. Why not use `Response.fail`?

File: dashboard/src/main/webapp/src/pages/serverstatus/NodeListPage.vue [hunk 2]
```diff
@@ -132,14 +152,19 @@ export default {
...
       } else if (router.currentRoute.value.name === 'decommissionedNodeList') {
+        isShowRemove.value = false
         getShuffleDecommissionedListPage()
       } else if (router.currentRoute.value.name === 'unhealthyNodeList') {
+        isShowRemove.value = false
         getShuffleUnhealthyListPage()
       } else if (router.currentRoute.value.name === 'lostNodeList') {
+        isShowRemove.value = true
         getShuffleLostListPage()
       }
```
Comment (xianjingfeng) [inline dashboard/src/main/webapp/src/pages/serverstatus/NodeListPage.vue:169]:
```suggestion
      isShowRemove.value = false
      if (router.currentRoute.value.name === 'activeNodeList') {
        getShuffleActiveNodesPage()
      } else if (router.currentRoute.value.name === 'decommissioningNodeList') {
        getShuffleDecommissioningListPage()
      } else if (router.currentRoute.value.name === 'decommissionedNodeList') {
        getShuffleDecommissionedListPage()
      } else if (router.currentRoute.value.name === 'unhealthyNodeList') {
        getShuffleUnhealthyListPage()
      } else if (router.currentRoute.value.name === 'lostNodeList') {
        isShowRemove.value = true
        getShuffleLostListPage()
      }
```

File: dashboard/src/main/webapp/src/utils/http.js [hunk 3]
```diff
@@ -48,6 +48,21 @@ const http = {
...
+  delete(url, params, headers, fontBackFlag) {
+    if (fontBackFlag === 0) {
+      // The system obtains the address of the Coordinator to be accessed from global variables.
+      const currentServerStore = useCurrentServerStore()
+      if (typeof headers !== 'undefined') {
+        headers.targetAddress = currentServerStore.currentServer
+      } else {
+        headers = {}
+        headers.targetAddress = currentServerStore.currentServer
+      }
```
[the remark this answers is not in the document]
Comment (yl09099) [reply dashboard/src/main/webapp/src/utils/http.js]: The points you mentioned have been revised.@xianjingfeng

**Review by xianjingfeng — APPROVED** [review]
LGTM

**Comment by rickyma** [comment]
Merged to master. Thanks @yl09099 @xianjingfeng

**Pull request merged**